src/visit.ts

```diff
@@ -480,3 +490,29 @@ export function visitEachChild<T extends FunctionlessNode>(
   }
   return assertNever(node);
 }
+
+/**
+ * Like {@link visitEachChild} but it only visits the statements of a block.
+ *
+ * Provides the hoist function that allows hoisting expressions into variable statements above the current statement.
+ */
+export function visitBlock(
+  block: BlockStmt,
+  cb: (stmt: Stmt, hoist: (expr: Expr) => Expr) => Stmt,
+  nameGenerator: DeterministicNameGenerator
+): BlockStmt {
+  return visitEachChild(block, (stmt) => {
+    const nestedTasks: FunctionlessNode[] = [];
+    function hoist(expr: Expr): Expr {
```

Comment:
Should the return type be an Identifier?

Reply:
ohh, yeah, it could be, it was more complex at one point.